src/gui/torrentoptionsdialog.cpp

```diff
@@ -343,8 +454,79 @@ int TorrentOptionsDialog::getSeedingTime() const
     return m_ui->spinTimeLimit->value();
 }
 
+void TorrentOptionsDialog::handleCategoryChanged(const int index)
+{
+    Q_UNUSED(index);
+
+    if (m_ui->checkAutoTMM->checkState() == Qt::Checked)
+    {
+        if (!m_allSameCategory && (m_ui->comboCategory->currentIndex() == 0))
+        {
+            m_ui->savePath->setSelectedPath(QLatin1String(""));
+        }
+        else
+        {
+            const QString savePath = BitTorrent::Session::instance()->categorySavePath(m_ui->comboCategory->currentText());
+            m_ui->savePath->setSelectedPath(Utils::Fs::toNativePath(savePath));
+        }
+    }
+
+    if (!m_allSameCategory && (m_ui->comboCategory->currentIndex() == 0))
+    {
+        m_ui->comboCategory->clearEditText();
+        m_ui->comboCategory->lineEdit()->setPlaceholderText(m_currentCategoriesString);
+    }
+    else
+    {
+        m_ui->comboCategory->lineEdit()->setPlaceholderText(QLatin1String(""));
+    }
+}
+
+void TorrentOptionsDialog::handleTMMChanged()
+{
+    if (m_ui->checkAutoTMM->checkState() == Qt::Unchecked)
+    {
+        m_ui->labelSavePath->setEnabled(true);
+        m_ui->savePath->setEnabled(true);
+        m_ui->savePath->setSelectedPath(Utils::Fs::toNativePath(m_initialValues.savePath));
+    }
+    else
+    {
+        m_ui->labelSavePath->setEnabled(false);
+        m_ui->savePath->setEnabled(false);
+        if (m_ui->checkAutoTMM->checkState() == Qt::Checked)
+        {
+            if (!m_allSameCategory && (m_ui->comboCategory->currentIndex() == 0))
+            {
+                m_ui->savePath->setSelectedPath(QLatin1String(""));
+            }
+            else
+            {
+                const QString savePath = BitTorrent::Session::instance()->categorySavePath(m_ui->comboCategory->currentText());
+                m_ui->savePath->setSelectedPath(Utils::Fs::toNativePath(savePath));
+            }
+        }
+        else // partially checked
+        {
+            m_ui->savePath->setSelectedPath(QLatin1String(""));
+        }
+    }
+}
+
 void TorrentOptionsDialog::handleRatioTypeChanged()
 {
+    if ((m_initialValues.ratio == MIXED_SHARE_LIMITS) || (m_initialValues.seedingTime == MIXED_SHARE_LIMITS))
+    {
+        QAbstractButton *currentRadio = m_ui->buttonGroup->checkedButton();
+        if ((currentRadio == m_previousRadio) && currentRadio)
```

Comment:
I would swap the predicates (check for nullptr first): `if (currentRadio && (currentRadio == m_previousRadio))`